### ISSUE TITLE
linux sys_read bug

### ISSUE THREAD

**Issue by dragazo** [dragazo/CSX64-cpp#1]
To simulate the real `sys_read`, csx64 is supposed to pause for input *(for a console, at least)* and only read as much as the user enters. The raw `istream::read()` function forces it to fill the buffer if the stream is interactive, so `CSX64::smart_readsome()` in `Utility.h` fixes this by wrapping several calls to `istream::readsome()`. This works fine in Windows, but in Linux `istream::readsome()` seems to always return zero (no data read)...

In short: `Utility.h: CSX64::smart_readsome()` is broken.

**Comment by dragazo** [comment]
I was hoping it wouldn't have to be refactored, but that ended up being the only way to do it.
